### DISCUSSION TITLE
How to set project root dir?

### DISCUSSION THREAD

**Discussion by matheusfillipe** [neoclide/coc.nvim#4899] [Q&A]
I ask this for: https://github.com/ThePrimeagen/git-worktree.nvim
How can I change `workspace.workspaceFolders` on the fly and restart lsp? I think that should be enough but can't figure out how to manually change the root path of the lsp.

**Comment by fannheyward** [comment]
Try `CocRestart`, coc.nvim will detect workspace root, restart language server.

**Comment by matheusfillipe** [comment]
I tried that but it doesn't change the root. workspace.workspaceFolders is still at the wrong root. I still have buffers open in both worktrees. Is there no way to just manually set it? Does it look for the current buffer or what does it look for?

**Reply by fannheyward** [reply]
>  look for the current buffer 

Yes, coc.nvim uses buffer document file path to resolve workspace root, no way to manually set it by now.

**Comment by matheusfillipe** [comment]
Is coc.nvim looking for the upper most match for rootPatterns? I have configured:

```json
  "languageserver": {
      "python": {
        "command": "jedi-language-server",
        "filetypes": ["py"],
        "rootPatterns": [".git", "pyproject.toml", "setup.py", "git/"]
      }
  }
```

My project has a `pyproject.toml` and `setup.py` and root dir. When I create a worktree it creates a new sub inside my project's root which contains all these files and a `.git` *file*, not folder.
Trying to reload the already running neovim instance in a buffer inside the worktree with with `CocRestart` doesn't update `workspace.workspaceFolders` but if I i open a fresh new instance inside the workdir the root path will be inside the worktree.
